sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoin.scala

```diff
@@ -59,19 +57,17 @@ case class BroadcastHashJoin(
   override def requiredChildDistribution: Seq[Distribution] =
     UnspecifiedDistribution :: UnspecifiedDistribution :: Nil
 
-  @transient
-  private val broadcastFuture = future {
+  private lazy val broadcastedRelation = {
```

Comment:
There was a noticeable speed-up in TPC-DS from broadcasting in parallel and I think this removes that.  Perhaps we should both use a `lazy val` and a `future`.  We can have the prepare phase make sure the future all get invoked before running.